txtorcon/endpoints.py

```diff
@@ -404,7 +404,8 @@ def info_event(msg):
             self.config, self.hidden_service_dir,
```

Comment:
I think creating the HiddenService instance should go inside the "if" as well -- otherwise the instance will change every time we call listen(). Thanks!